mlfinlab/microstructural_features/feature_generator.py

```diff
@@ -0,0 +1,301 @@
+"""
+Inter-bar feature generator which uses trades data and bars index to calculate inter-bar features
+"""
+
+import pandas as pd
+import numpy as np
+from mlfinlab.microstructural_features.entropy import get_shannon_entropy, get_plug_in_entropy, get_lempel_ziv_entropy
+from mlfinlab.microstructural_features.encoding import encode_array
+from mlfinlab.microstructural_features.second_generation import get_trades_based_kyle_lambda, \
+    get_trades_based_amihud_lambda, get_trades_based_hasbrouck_lambda
+from mlfinlab.microstructural_features.misc import get_avg_tick_size, vwap
+from mlfinlab.microstructural_features.encoding import encode_tick_rule_array
+
+
+# pylint: disable=too-many-instance-attributes
+
+def _crop_data_frame_in_batches(df: pd.DataFrame, chunksize: int):
+    # pylint: disable=invalid-name
+    """
+    Splits df into chunks of chunksize
+
+    :param df: (pd.DataFrame) to split
+    :param chunksize: (Int) number of rows in chunk
+    :return: (list) of chunks (pd.DataFrames)
+    """
+    generator_object = []
+    for _, chunk in df.groupby(np.arange(len(df)) // chunksize):
+        generator_object.append(chunk)
+    return generator_object
```

Comment:
I saw this one in data structures -> base_bars.py, maybe use this function from some kind of util module?